src/codegen/function.rs

```diff
@@ -23,9 +23,30 @@ use std::{
     result::Result as StdResult,
 };
 
+pub fn get_must_use_if_needed(
+    parent_type_id: Option<TypeId>,
+    analysis: &analysis::functions::Info,
+    in_trait: bool,
+    only_declaration: bool,
+    comment_prefix: &str,
+) -> Option<String> {
+    if let Some(parent_type_id) = parent_type_id {
+        if analysis.kind == library::FunctionKind::Method && (!in_trait || only_declaration) {
+            let outs = out_parameter_types(analysis);
+            if let [return_type] = *outs.as_slice() {
+                if parent_type_id == return_type {
```

Comment:
```suggestion
            if [parent_type_id] == *outs.as_slice() {
```

Maybe?

Reply:
Wait that works? O.o

Reply:
No it doesn't work. You can check it with:

```rust
fn check(s: &[String], matcher: &String) {
    if let [matcher] = s {
        eprintln!("yolo!");
    }
}

fn main() {
    let matcher = "b".to_owned();
    check(&vec!["a".to_owned()], &matcher);
    check(&vec!["b".to_owned()], &matcher);
}
```

Reply:
Ah indeed, updating then!